docs/source/api/apollo-server.md

```diff
@@ -342,20 +342,57 @@ addMockFunctionsToSchema({
    to standard error. Specify this function to process errors in a different
    way.
+   
+* `sendVariableValues`: { transform: (options: { variables: Record<string, any>, operationString?: string } ) => Record<string, any> }
+                     | { exceptNames: Array&lt;String&gt; }
+                     | { includeNames: Array&lt;String&gt; }
+                     | { sendNone: true }
+                     | { sendAll: true }
+    
+    By default, Apollo Server does not send the values of any GraphQL variables to Apollo's servers, because variable values often contain the private data of your app's users. If you'd like variable values to be included in traces, set this option. This option can take several forms:
+    
+    - { sendNone: true }: blocklist all variable values (DEFAULT)
+    - { sendAll: true }: safelist all variable values
+    - { transform: ... }: a custom function for modifying variable values. Keys added by the custom function will be removed, and keys removed will be added back with an empty value. 
+    - { exceptNames: ... }: a case-sensitive list of names of variables whose values should not be sent to Apollo servers
+    - { includeNames: ... }: a case-sensitive list of names of variables whose values will be sent to Apollo Servers
+
+   Defaults to blocklisting all variable values if both this parameter and the to-be-deprecated `privateVariables` are not set. 
+   The report will indicate each private variable key whose value was redacted by { sendNone: true } or { exceptNames: [...] }.
+   
+*  `privateVariables`: Array&lt;String&gt; | boolean
 
-*  `privateVariables`: Array<String> | boolean
-
-   A case-sensitive list of names of variables whose values should not be sent
-   to Apollo servers, or 'true' to leave out all variables. In the former
-   case, the report will indicate that each private variable was redacted in
-   the latter case, no variables are sent at all.
-
-*  `privateHeaders`: Array<String> | boolean
-
-   A case-insensitive list of names of HTTP headers whose values should not be
-   sent to Apollo servers, or 'true' to leave out all HTTP headers. Unlike
-   with privateVariables, names of dropped headers are not reported.
+   <!--- TODO(helen): update the version number here --->
+   DEPRECATING IN VERSION XX.XX.XX, to be replaced by the option `sendVariableValues`, which supports the same
+   functionalities but allows for more flexibility. Passing an array into `privateVariables` is equivalent to
+   passing in `{ exceptNames: array } ` to `sendVariableValues`, and passing in `true` or `false` is equivalent
+   to passing ` { sendNone: true } ` or ` { sendAll: true }`, respectively.
+   
+   NOTE: An error will be thrown if both this deprecated option and its replacement, `sendVariableValues` are defined.
+    
+* `sendHeaders`: { exceptNames: Array&lt;String&gt; } | { includeNames: Array&lt;String&gt; } | { sendAll: boolean } | { sendNone: boolean }
```

Comment:
Seems a little surprising to not include `transform` here too, though I suppose the main use case of transform is to tweak nested values which isn't relevant here.

Reply:
I didn't include that option because I wasn't sure there was a need for it, maybe this is something that we could get feedback on? But otherwise I think this would be a relatively simple extension to make!